include/grpc/grpc_security.h

```diff
@@ -341,6 +377,39 @@ GRPCAPI grpc_server_credentials *grpc_ssl_server_credentials_create_ex(
     grpc_ssl_client_certificate_request_type client_certificate_request,
     void *reserved);
 
+typedef struct grpc_ssl_server_credentials_options
+    grpc_ssl_server_credentials_options;
+
+/** Creates an options object using a certificate config. Use this method when
+   the certificates and keys of the SSL server will not change during the
+   server's lifetime.
+   - Takes ownership of the certificate_config parameter. */
+GRPCAPI grpc_ssl_server_credentials_options *
+grpc_ssl_server_credentials_create_options_using_config(
+    grpc_ssl_client_certificate_request_type client_certificate_request,
+    grpc_ssl_server_certificate_config **certificate_config);
```

Comment:
Why a double pointer here? We're not returning a config: we're just passing ownership so a single pointer will do.

Reply:
@jboeuf i mentioned the double-pointer as a possible approach here https://github.com/grpc/grpc/commit/8d975e3#commitcomment-24969012

it helps make the ownership transfer a little more natural, e.g., i think kind of like a `uniuqe_ptr` in c++11... without the double pointer you would need something like:
```
type* ptr = ...;
... = func(ptr);
if (<ownership was transferred, i.e., by examining returned value of func()>) {
  ptr = NULL;
}
...
free(ptr);
}
```

with the double pointer, then `func()` could nullifies the `ptr` on success, and the caller here doesn't have to do the "if ownership was transferred, then ptr = NULL" stuff.